util/src/dbutil.cpp

```diff
@@ -128,6 +128,38 @@ void sqliteRegexImpl(
 #endif
 
 /**
+ * This function creates a postgres database if the database doesn't exists.
+ */
+void createPsqlDatbase(const std::string& connStr_, const std::string dbName_)
+{
+  std::size_t colonPos = connStr_.find(':');
+  std::string database = connStr_.substr(0, colonPos);
+  std::string options  = connStr_.substr(colonPos + 1);
+
+  std::vector<std::string> odbOpts = createOdbOptions(options);
+  char** cStyleOptions = createCStyleOptions(odbOpts);
+
+  std::shared_ptr<odb::database> db;
+  int size = odbOpts.size();
+  db.reset(new odb::pgsql::database(size, cStyleOptions),
+    [](odb::database*){});
+
+  odb::connection_ptr connection = db->connection();
+  try
+  {
+    std::string createCmd = "CREATE DATABASE " + dbName_
+      + " ENCODING = 'SQL_ASCII'"
+      + " LC_COLLATE='C'"
+      + " TEMPLATE template0;";
+
+    connection->execute(createCmd);
+
+    LOG(info) << "Creating database: " << dbName_;
+  }
+  catch (...) {}
```

Comment:
How is it handled if the database already exists?
How is it handled if the database connection fails (e.g. the connected user does not have the relevant permission grant)? Without a database, operations cannot continue.